src/components/cesium_map/cesium_UI.js

```diff
@@ -389,15 +382,12 @@ class CesiumMap extends React.Component {
   // https://medium.com/@garrettmac/reactjs-how-to-safely-manipulate-the-dom-when-reactjs-cant-the-right-way-8a20928e8a6
   // manipulate Dom outside the react model
   shouldComponentUpdate(nextProps) {
-    // update point primitive based on searchFields
-    const sf1 = JSON.stringify(nextProps.newSearchFields);
-    const sf2 = JSON.stringify(this.props.newSearchFields);
-    if (sf1 !== sf2) {
-      // clear all element in cesium
-      searchFields = nextProps.newSearchFields;
-      clearBoundingBox(true);
-      this.updatePrimitive(viewer.currentView.latitude, viewer.currentView.longitude);
-    };
+    // this method will be called when the search field facet changed 
+    // clear all element in cesium
+    searchFields = nextProps.newSearchFields;
+    clearBoundingBox(true);
+    // update the point layer
+    this.updatePrimitive(viewer.currentView.latitude, viewer.currentView.longitude);
 
```

Comment:
This is where we update the point layer of the map. This is only called when the number of documents of query result changed due to new search facet field added.